src/main/webapp/view/transcript/transcript.js

```diff
@@ -171,3 +173,43 @@ function scrollToTopOfTranscript(transcriptLine) {
   const ulElementOffset = transcriptLine.parentElement.offsetTop;
   transcriptContainer.scrollTop = transcriptLine.offsetTop - ulElementOffset;
 }
+
+/**
+ * Returns the next transcript line based on `currentTimeMs`.
+ */
+function getNextTranscriptLine(currentTimeMs) {
+  const nextTranscript = currentTranscriptLine.nextElementSibling;
+  // Video is playing normally.
+  if (isWithinTimeRange(currentTimeMs, nextTranscript)) {
+    return nextTranscript;
+  }
+  return findClosestTranscriptLine(currentTimeMs);
+}
+
+/**
+ * Searches for and returns the closest transcript line
+ * based on `timeMs`.
+ */
+function findClosestTranscriptLine(timeMs) {
```

Comment:
should 'timeMs' be renamed 'videoTimeMs'?

Reply:
I think it might be better to keep it as `timeMs` because this function should also work on a time greater than the video length. I'm not sure if I want to limit the scope by calling it `videoTimeMs`